StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1214StaticReadonlyElementsMustAppearBeforeStaticNonReadonlyElements.cs

```diff
@@ -43,7 +46,61 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(this.AnalyzeClass, SyntaxKind.ClassDeclaration);
+            context.RegisterSyntaxNodeAction(this.AnalyzeStruct, SyntaxKind.StructDeclaration);
+        }
+
+        private void AnalyzeStruct(SyntaxNodeAnalysisContext context)
+        {
+            var structDeclaration = (StructDeclarationSyntax)context.Node;
+
+            AnalyzeType(context, structDeclaration);
+        }
+
+        private void AnalyzeClass(SyntaxNodeAnalysisContext context)
+        {
+            var classDeclaration = (ClassDeclarationSyntax) context.Node;
+
+            AnalyzeType(context, classDeclaration);
+        }
+
+        private static void AnalyzeType(SyntaxNodeAnalysisContext context, TypeDeclarationSyntax typeDeclaration)
+        {
+            var staticFields = typeDeclaration.Members
+                .OfType<BaseFieldDeclarationSyntax>()
+                .Where(f => f.Modifiers.Any(m => m.IsKind(SyntaxKind.StaticKeyword)))
+                .ToList();
+
+            var endLocationsOfStaticNonReadonlyFields =
+                staticFields.Where(f => f.Modifiers.Any(m => m.IsKind(SyntaxKind.ReadOnlyKeyword)) == false)
+                    .Select(f => f.GetLocation().GetLineSpan())
+                    .Where(l => l.IsValid)
+                    .Select(l => l.EndLinePosition)
+                    .ToList();
+
+            var readonlyFieldLocations =
+                staticFields.Where(f => f.Modifiers.Any(m => m.IsKind(SyntaxKind.ReadOnlyKeyword)))
+                    .Where(f => f.GetLocation().GetLineSpan().IsValid)
+                    .Where(f => f.Declaration != null && f.Declaration.Variables.Any())
+                    .Select(f => new {
+                        FieldLocations = f.Declaration.Variables
+                                                      .Where(v => !v.IsMissing && !v.Identifier.IsMissing)
+                                                      .Select(v => v.Identifier.GetLocation())
+                                                      .ToList(),
+                        FieldEndLinePosition = f.GetLocation().GetLineSpan().EndLinePosition
+                    })
+                    .ToList();
+
+            foreach (var location in readonlyFieldLocations)
+            {
+                if (endLocationsOfStaticNonReadonlyFields.Any(e => e < location.FieldEndLinePosition))
```

Comment:
:exclamation: You don't need all `static` non-`readonly` fields. You only need the first one. Fixing this will change the search algorithm from O(n²) to O(n).